edv4ldez/react-apprenticeship-capstone-2
# Capston 2 NasaApp

## NasaApp
```
It's a react application implementing styled-components, context and performance corresponding to the topics of Wizeline's React Apprenticeship program. I have worked designing and creating reusable components, getting and using data from Nasa APOD and applying TDD.
- The initial page contains the image or video of the day.
- You can select a date to search an image.
- To run: npm i; npm start ; To run tests: npm run test
```

## Type of Change
```
## How should this be tested?
```
Run: npm run test
- Testing useDataFetch to validate the cases if we get data or error.
<img width="590" alt="Screen Shot 2022-01-25 at 14 11 47" src="https://user-images.githubusercontent.com/97318658/151063551-777696fa-e905-47d8-8e37-950b0109bc78.png">

- Testing getDateToday to verify that the algorithm is returning the date of today.
<img width="590" alt="Screen Shot 2022-01-25 at 14 11 47" src="https://user-images.githubusercontent.com/97318658/151063990-813d36d2-77e2-4a44-ae0a-07487971f046.png">

- Testing DatePicker.component to validate the initial state and the state when we submit the form.
<img width="623" alt="Screen Shot 2022-01-25 at 14 13 01" src="https://user-images.githubusercontent.com/97318658/151064305-55d91f5e-24b0-4540-8459-5649c55ccc75.png">

-Testing ImageCard.component to verify the render of the image or error if we don't get data from NASA APOD.
<img width="600" alt="Screen Shot 2022-01-25 at 14 13 25" src="https://user-images.githubusercontent.com/97318658/151064672-9c102ffb-b15a-4570-bb94-471dcf90a67f.png">
```
## What is the deliverable?
```
- Final deliverable
```
## Screenshots and links
```
- Desktop screen and initial state.
<img width="1607" alt="Screen Shot 2022-01-25 at 14 14 50" src="https://user-images.githubusercontent.com/97318658/151065088-98c5fe46-eca3-4af5-8c8b-cea666dca38d.png">

- Middle screen and initial state.
<img width="932" alt="Screen Shot 2022-01-25 at 14 15 48" src="https://user-images.githubusercontent.com/97318658/151065167-0454df7d-5b6d-4773-a1c5-2d4f0694d720.png">
<img width="1607" alt="Screen Shot 2022-01-25 at 14 14 50" src="https://user-images.githubusercontent.com/97318658/151065234-8235ce8f-4b06-43d9-b1b7-71bd8dbc16a7.png">

- Phone screen and initial state.
<img width="538" alt="Screen Shot 2022-01-25 at 14 16 28" src="https://user-images.githubusercontent.com/97318658/151065249-2060ac81-da94-48b6-a472-adbd428a67a8.png">

- Selecting a date.
<img width="533" alt="Screen Shot 2022-01-25 at 14 17 42" src="https://user-images.githubusercontent.com/97318658/151071405-7f7860d9-ca7c-497e-af0f-63d5b27b389d.png">

- Getting image or video from NASA APOD
<img width="547" alt="Screen Shot 2022-01-25 at 14 18 36" src="https://user-images.githubusercontent.com/97318658/151065378-c085b621-300b-4d69-971a-68814aa5c170.png">

- Tests output.
<img width="349" alt="Screen Shot 2022-01-25 at 15 51 25" src="https://user-images.githubusercontent.com/97318658/151069371-8bbb75c5-3943-430f-a9c7-49d5c3ad310a.png">

[Capston 2 NASA APOD](https://youthful-aryabhata-ee129a.netlify.app/)

## Checklist
